plotly/tools.py

```diff
@@ -15,14 +15,16 @@
 import six
 
 import math
+import scipy
 
+import scipy.cluster.hierarchy as sch
```

Comment:
We'll have to protect these imports into the actual dendrogram function. Not everyone using Plotly will have these packages installed, and it's a dependency that we would like to avoid for as long as possible! However, it's OK for the dendrogram class to use these functions, you'll just have to protect these imports. Here is how we import numpy, scipy and scipy.cluster.hierarcy can use similar logic:

Protected import: https://github.com/plotly/python-api/blob/master/plotly/tools.py#L48-L52
Raise exception if the import fails and they try to use the function that requires it: https://github.com/plotly/python-api/blob/master/plotly/tools.py#L1534-L1535